wdl2cwl/main.py

```diff
@@ -36,784 +21,788 @@
+            outputs=cwl_outputs,
+            cwlVersion="v1.2",
+            baseCommand=base_command,
         )
 
-        raw_input_value = i[2].replace('"', "")
-        input_value: Union[str, bool, int, float] = ""
-
-        if "boolean" in input_type:
-            input_value = bool(raw_input_value.lower() == "true")
-        elif "int" in input_type:
-            input_value = int(raw_input_value)
-        elif "float" in input_type:
-            input_value = float(raw_input_value)
+    def get_time_minutes_requirement(
+        self, time_minutes: WDL.Expr.Get
+    ) -> Union[str, int]:
+        """Produce the time limit expression from WDL runtime time minutes."""
+        if isinstance(time_minutes, (WDL.Expr.Int, WDL.Expr.Float)):
+            literal = self.get_wdl_literal(time_minutes.literal)  # type: ignore
+            return literal * 60  # type: ignore
+        time_minutes_str = self.get_expr(time_minutes)
+        return f"$({time_minutes_str} * 60)"
+
+    def get_outdir_requirement(
+        self, outdir: Union[WDL.Expr.Get, WDL.Expr.Apply]
+    ) -> int:
+        """Produce the memory requirement for the output directory from WDL runtime disks."""
+        # This is yet to be implemented. After Feature Parity.
+        return int(self.get_wdl_literal(outdir.literal)) * 1024  # type: ignore
+
+    def get_input(self, input_name: str) -> str:
+        """Produce a consise, valid CWL expr/param reference lookup string for a given input name."""
+        if valid_js_identifier.match(input_name):
+            return f"inputs.{input_name}"
+        return f'inputs["{input_name}"]'  # pragma: no cover
+
+    def get_memory_requirement(
+        self, memory_runtime: Union[WDL.Expr.Ident, WDL.Expr.Get, WDL.Expr.String]
+    ) -> Union[str, float]:
+        """Translate WDL Runtime Memory requirement to CWL Resource Requirement."""
+        if isinstance(memory_runtime, WDL.Expr.String):
+            ram_min_literal = self.get_memory_literal(memory_runtime)
+            return ram_min_literal
+        ram_min = self.get_expr_name(memory_runtime.expr)  # type: ignore
+        return self.get_ram_min_js(ram_min, "")
+
+    def get_memory_literal(self, memory_runtime: WDL.Expr.String) -> float:
+        """Get the literal value for memory requirement with type WDL.Expr.String."""
+        if memory_runtime.literal is None:
+            _, placeholder, unit, _ = memory_runtime.parts
+            value_name = self.get_expr_get(placeholder.expr)  # type: ignore
+            return self.get_ram_min_js(value_name, unit.strip())  # type: ignore
+
+        ram_min = self.get_expr_string(memory_runtime)[1:-1]
+        unit = re.search(r"[a-zA-Z]+", ram_min).group()  # type: ignore
+        value = float(ram_min.split(unit)[0])
+
+        if unit == "KiB":
+            memory = value / 1024
+        elif unit == "MiB":
+            memory = value
+        elif unit == "GiB":
+            memory = value * 1024
```

Comment:
+1

@Th3nn3ss for this one, my approach would be to

1. write a simple pytest unit test
2. since there is an `if` block being tested, you will probably need several test cases to cover all the `if` branches, so I would start with a parametrized test case, e.g. https://github.com/common-workflow-lab/wdl-cwl-translator/blob/44410610d35849a2ccf30652fe302bb5c4bbf032/wdl2cwl/tests/test_cwl.py#L122-L137
3. instead of calling `wdl.convert`, my test would create an instance of the Converter, and then call `get_memory_literal`

On my IDE, I would then run that test separately, sometimes debugging to make sure it is covering the branches the way I expect it. Once all tests are written and I am happy with, then I would run the tests with coverage, and use `make` and/or `tox`. Then update the pull request.

We can have a one-on-one chat, maybe using the VSCode to share your screen, and try to work on this one together if you'd like.

A test like this one should take between 10 minutes and 1 hour max I think. Maybe a little more if it is too complicated to create `WDL.Expr.String` objects programmatically. If you think it will take much longer than that, this one-on-one session can help you avoid getting stuck :slightly_smiling_face: Just let me know :+1: